jsettlers.graphics/src/main/java/jsettlers/graphics/map/draw/MapObjectDrawer.java

```diff
@@ -430,62 +432,138 @@ public void draw(IMovable movable) {
 	}
 
 	private void playMovableSound(IMovable movable) {
-		if (!movable.isSoundPlayed()) {
-			final EMovableAction action = movable.getAction();
-			if (action == EMovableAction.ACTION1) {
-				playSoundAction1(movable.getMovableType(), movable.getPos());
-				movable.setSoundPlayed();
-			} else if (action == EMovableAction.ACTION2) {
-				playSoundAction2(movable.getMovableType(), movable.getPos());
-				movable.setSoundPlayed();
-			}
+		if (movable.isSoundPlayed()) {
+			return;
 		}
-	}
-
-	private void playSoundAction1(EMovableType type, ShortPoint2D position) {
-		switch (type) {
-		case BRICKLAYER:
-			sound.playSound(1, 1, position);
-			break;
-		case LUMBERJACK:
-			sound.playSound(0, 1, position);
-			break;
-		case SAWMILLER:
-			sound.playSound(5, 1, position);
-			break;
-		case STONECUTTER:
-			sound.playSound(3, 1, position);
-			break;
-		case DIGGER:
-			sound.playSound(2, 1, position);
-			break;
-		case SMITH:
-			sound.playSound(6, 1, position);
-			break;
-		case FARMER:
-			sound.playSound(12, 1, position);
-			break;
-		case SWORDSMAN_L1:
-		case SWORDSMAN_L2:
-		case SWORDSMAN_L3:
-			sound.playSound(30, 1, position);
-			break;
-		case BOWMAN_L1:
-		case BOWMAN_L2:
-		case BOWMAN_L3:
-			sound.playSound(33, 1, position);
-			break;
-		case CHARCOAL_BURNER:
-			sound.playSound(45, 1, position);
+		int soundNumber = -1;
+		float delay = movable.getMoveProgress();
+		switch (movable.getAction()) {
+		case ACTION1:
+			switch (movable.getMovableType()) {
+			case LUMBERJACK:
+				if (delay > .8) {
+					soundNumber = 0;
+				}
+				break;
+			case BRICKLAYER:
+				if (delay > .7) {
+					soundNumber = 1;
+				}
+				break;
+			case DIGGER:
+				if (delay > .6) {
+					soundNumber = 2;
+				}
+				break;
+			case STONECUTTER:
+				if (delay > .8) {
+					soundNumber = 3;
+				}
+				break;
+			case SAWMILLER:
+				if (delay > .2) {
+					soundNumber = 5;
+				}
+				break;
+			case SMITH:
+				if (delay > .7) {
+					soundNumber = 6;
+				}
+				break;
+			case FARMER:
+				if (delay > .8) {
+					soundNumber = 9;
+				}
+				break;
+			case FISHERMAN:
+				if (delay > .8) {
+					soundNumber = 16;
+				}
+				break;
+			case DOCKWORKER:
+				if (delay > .8) {
+					soundNumber = 20;
+				}
+				break;
+			case HEALER:
+				if (delay > .8) {
+					soundNumber = 21;
+				}
+				break;
+			case GEOLOGIST: // TODO: should also check grid.getResourceAmountAt(x, y)
+				if (sound.random.nextInt(256) == 0) {
```

Comment:
why are you doing the random here? In only one of 256 cases, you play the sound?

Reply:
else the geologists would shout far too often